prisma/schema.prisma

```diff
@@ -11,23 +11,56 @@ model Site {
   id      String   @id @default(cuid())
   name    String
   types   String[]
+  lat     Float
+  lng     Float
   city    String
   country String
   radius  Int
-  User    User?    @relation(fields: [userId], references: [id])
+  user    User?    @relation(fields: [userId], references: [id])
   userId  String?
-  lat     Float
-  lng     Float
+}
+
+model Account {
+  id                 String  @id @default(cuid())
+  userId             String
+  type               String
+  provider           String
+  providerAccountId  String
+  refresh_token      String? @db.Text
+  access_token       String? @db.Text
+  expires_at         Int?
+  refresh_expires_in Int?
+  not_before_policy  Int?
+  token_type         String?
+  scope              String?
+  id_token           String? @db.Text
+  session_state      String?
+  user               User    @relation(fields: [userId], references: [id])
+
+  @@unique([provider, providerAccountId])
+}
+
+model Session {
+  id           String   @id @default(cuid())
+  sessionToken String   @unique
+  userId       String
+  expires      DateTime
+  user         User     @relation(fields: [userId], references: [id])
 }
 
 model User {
-  id                     String   @id @default(cuid())
+  id                     String    @id @default(cuid())
   name                   String
-  password               String
-  phone                  String
-  allowPushNotifications Boolean  @map(name: "allow_push_notifications")
-  allowSMSNotifications  Boolean  @map(name: "allow_sms_notifications")
+  password               String?
+  phone                  String?
+  email                  String?   @unique
+  emailVerified          DateTime?
+  image                  String?
+  allowPushNotifications Boolean?  @map(name: "allow_push_notifications")
+  allowSMSNotifications  Boolean?  @map(name: "allow_sms_notifications")
   sites                  Site[]
-  createdAt              DateTime @default(now()) @map(name: "created_at")
-  updatedAt              DateTime @updatedAt @map(name: "updated_at")
+  createdAt              DateTime  @default(now()) @map(name: "created_at")
+  updatedAt              DateTime  @updatedAt @map(name: "updated_at")
+  accounts               Account[]
```

Comment:
Why does the user have several accounts?

Reply:
I believe that is because a user can be logged in from different devices, resulting in the need for multiple sessions and accounts

Reply:
Aah, I understand. Ideally Keycloak should handle all of this for us. Maybe see if this is something Keycloak already supports out of the box

Reply:
Also, we are currently not planning on adding more providers than KC. And if we do I think it's fair enough to just say each user can only have one login-provider